timelock-agent/src/main/java/com/palantir/timelock/config/TimeLockInstallConfiguration.java

```diff
@@ -18,12 +22,27 @@
 @JsonSerialize(as = ImmutableTimeLockInstallConfiguration.class)
 @Value.Immutable
 public interface TimeLockInstallConfiguration {
-    PaxosInstallConfiguration paxos();
+    Optional<PaxosInstallConfiguration> optionalPaxosConfig();
+
+    Optional<KeyValueServiceConfig> optionalKvsConfig();
 
     ClusterConfiguration cluster();
 
     @Value.Default
     default AsyncLockConfiguration asyncLock() {
         return ImmutableAsyncLockConfiguration.builder().build();
     }
+
+    @Value.Default
+    default PaxosInstallConfiguration paxos() {
```

Comment:
This feels dangerous. I get the sense that any time this value isn't populate it shouldn't be referenced at all and thus passing the default value through feels scary. I'd rather this be some sort of `orElseThrow` if that exists.

Reply:
Also not clear to me we need this; I think the dev-break to Optional is fine here.